resource.go

```diff
@@ -158,7 +169,25 @@ func (r *resource) pushMetrics(metrics []byte, dst string, wg *sync.WaitGroup) {
 		}).Error("Failed to push metrics.")
 		return
 	}
-	resp.Body.Close()
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.WithFields(logrus.Fields{
+			"error":         err.Error(),
+			"resource_name": r.name,
+			"resource_url":  r.resURL,
+		}).Error("Failed to read response body.")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		logger.WithFields(logrus.Fields{
+			"body":          body,
+			"status":        resp.StatusCode,
+			"resource_name": r.name,
+			"resource_url":  r.resURL,
+		}).Error("Got non-OK status code.")
```

Comment:
A distinctive error message might come in handy, so that you can filter for it in the logs / create alerting (eg. 'Got non-OK status code pushing metrics.'